hazelcast-jet-core/src/main/java/com/hazelcast/jet/impl/pipeline/transform/TimestampTransform.java

```diff
@@ -41,10 +43,12 @@ public TimestampTransform(
 
     @Override
     public void addToDag(Planner p) {
+        PlannerVertex upstream = p.xform2vertex.get(this.upstream().get(0));
+        int localParallelism = determineLocalParallelism(upstream.v, upstream.v.getLocalParallelism());
         PlannerVertex pv = p.addVertex(
-                this, name(), localParallelism(), insertWatermarksP(eventTimePolicy)
+                this, name(), localParallelism, insertWatermarksP(eventTimePolicy)
         );
-        p.addEdges(this, pv.v);
+        p.dag.edge(between(upstream.v, pv.v).isolated());
```

Comment:
The right way to configure edge is to use p.addEdges(this, pv.v, edge -> edge.isolated()), this doesn't take into account multiple output ordinals.